src/composers/field.tsx

```diff
@@ -59,10 +62,11 @@ export function Field<Props>(WrappedComponent: IncomingField<Props>,
         }
 
         public render() {
+            const fieldState = this.props.form.getFieldState(this.getFieldName());
```

Comment:
Ehm, this is already what we're doing.. I just say we remove the `state` from the `Field` and `Listener` components and introduce a `this.forceUpdate()` within the `listenForFieldChange` callback. Then in the `render` we just fetch the values we need.